### DISCUSSION TITLE
How to copy builded c++ deps to temp dir created by cibuildwheel?

### DISCUSSION THREAD

**Discussion by MarshalX** [pypa/cibuildwheel#645] [Q&A] [answered]
As you can see in the screenshot, I have a dependencies folder that is used when building. When building, relative paths from the current directory are used. How do I copy the dependencies folder to a temporary folder? I do not see the corresponding placeholder for CIBW_BEFORE_BUILD. Only placeholder for package path ({package}).

My folder structure for understanding:
- tgcalls
  - Libraries
      - macos
          - ffmpeg
  - tgcalls
      - setup.py

>  pip wheel /Users/runner/work/tgcalls/tgcalls/tgcalls -w /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibuildwheel_vhh3uq9/built_wheel --no-deps

How to get `/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibuildwheel_vhh3uq9` path in CIBW_BEFORE_BUILD?

![image](https://user-images.githubusercontent.com/15520314/115597031-1478dc00-a2d9-11eb-8ff5-5cc30e180410.png)

**Comment by MarshalX** [comment]
### My temporary solution 

1. I added a new placeholder (temp_dir) for macOS to my cibuildwheel fork ([commit](https://github.com/MarshalX/cibuildwheel/commit/8d023bd2171a887e1139979d29731be1b9a52735))
2. Install cibuildwheel from git in my CI workflow
    - old row `pip install cibuildwheel`
    - new one `pip install git+https://github.com/MarshalX/cibuildwheel@temp-dir-macos-placeholder`
3. Add _CIBW_BEFORE_ALL_ with coping of folder with deps into temp dir
```
env:
    CIBW_BEFORE_ALL: mkdir {temp_dir}/../Libraries && cp -R $LibrariesPath {temp_dir}/../Libraries
```

I think the new placeholder is a good solution to the current problem. Let me know if I should finalize my feature and create a pull request

**Comment by Czaki** [comment, marked as answer]
I strongly suggest to move `setup.py` one level up or use the environment variable inside `setup.py` to identify the path to external library. 

You could use in `setup.py` construction like `os.environ.get("MY_LIBRARY_BASE_PATH", "..")`  

Did your code build locally with `pip wheel` command? I think not. If I good remember your fix works only because of the flat structure of tmp directories created by the command `pip wheel`. 

If you use environment variables in `setup.py` then `package` placeholder should be enough.

**Reply by MarshalX** [reply]
> Did your code build locally with `pip wheel` command? I think not. 

You are right.

> I strongly suggest to move `setup.py` one level up or use the environment variable inside `setup.py` to identify the path to external library.

Initially, I didn't want to edit the CMakeLists in order to use the cibuildwheel. But this solution is really better. That's exactly what I did. Dziękuję!
